src/menu/menu.service.ts

```diff
@@ -39,43 +38,34 @@ export class MenuService {
       }),
     ]);
 
-    if (modificationGroups?.length === 0) {
+    if (!modifications || modifications?.length === 0) {
```

Comment:
Why check !modifications?
modifications?.length is either number or undefined. 
Undefined will never be equal to 0.